drkafka/src/main/java/com/pinterest/doctorkafka/api/DecommissionBrokerApi.java

```diff
@@ -0,0 +1,53 @@
+package com.pinterest.doctorkafka.api;
+
+import com.pinterest.doctorkafka.DoctorKafka;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+
+@Path("/cluster/{clusterName}/broker/{brokerId}/admin/decommission")
+@Produces({MediaType.APPLICATION_JSON })
+@Consumes({MediaType.APPLICATION_JSON })
+public class DecommissionBrokerApi extends DoctorKafkaApi {
+
+  private static final Logger LOG = LogManager.getLogger(DecommissionBrokerApi.class);
+
+  public DecommissionBrokerApi(DoctorKafka drkafka) {
+    super(drkafka);
+  }
+
+  @GET
+  public boolean isBrokerDecommissioned(@PathParam("clusterName") String clusterName, @PathParam("brokerId") String brokerId) {
+    return checkAndGetBroker(clusterName, brokerId).isDecommissioned();
+  }
+
+  @PUT
+  public void decommissionBroker(@Context HttpServletRequest ctx,
```

Comment:
We should at least have metrics for decommissioning. 

In addition, I would recommend that email alerting be enabled for this.

Reply:
Added email alerting